sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/ClientSecretCredentialBuilder.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Fluent credential builder for instantiating a {@link ClientSecretCredentialBuilder}.
+ *
+ * @see ClientSecretCredentialBuilder
+ */
+public class ClientSecretCredentialBuilder extends AadCredentialBuilderBase<ClientSecretCredentialBuilder> {
+    private String clientSecret;
+
+    /**
+     * Sets the client secret for the authentication.
+     * @param clientSecret the secret value of the AAD application.
+     * @return the ClientSecretCredentialBuilder itself
+     */
+    public ClientSecretCredentialBuilder clientSecret(String clientSecret) {
```

Comment:
So far I have resisted the urge to have constructor arguments for the required fields, and just had constructors with no-args. I understand the desire for required args being in the constructor, but this can be a slippery slope and remove the readability benefits of a fluent builder pattern.

Reply:
Maybe I'm missing something but this is a fluent setter?

Reply:
Ah - my bad. I saw this as the constructor! Ignore me :-)